tests/Microsoft.Identity.Test.Unit/TelemetryTests/OTelInstrumentationTests.cs

```diff
@@ -60,7 +61,101 @@ public async Task AcquireTokenOTelTestAsync()
                 await AcquireTokenMsalClientExceptionAsync().ConfigureAwait(false);
 
                 s_meterProvider.ForceFlush();
-                VerifyMetrics(5, _exportedMetrics);
+                VerifyMetrics(5, _exportedMetrics, 2, 2);
+            }
+        }
+
+        [TestMethod]
+        [Description("AT in cache, needs refresh. AAD responds well to Refresh.")]
+        public async Task ProactiveTokenRefresh_ValidResponse_Async()
+        {
+            // Arrange
+            using (_harness = base.CreateTestHarness())
+            {
+                Trace.WriteLine("1. Setup an app");
+                CreateApplication();
+                TokenCacheHelper.PopulateCache(_cca.AppTokenCacheInternal.Accessor, addSecondAt: false);
+
+                Trace.WriteLine("2. Configure AT so that it shows it needs to be refreshed");
+                var refreshOn = TestCommon.UpdateATWithRefreshOn(_cca.AppTokenCacheInternal.Accessor).RefreshOn;
+
+                TokenCacheAccessRecorder cacheAccess = _cca.AppTokenCache.RecordAccess();
+
+                Trace.WriteLine("3. Configure AAD to respond with valid token to the refresh RT flow");
+                _harness.HttpManager.AddAllMocks(TokenResponseType.Valid_ClientCredentials);
+
+                // Act
+                Trace.WriteLine("4. ATS - should perform an RT refresh");
+                AuthenticationResult result = await _cca.AcquireTokenForClient(TestConstants.s_scope)
```

Comment:
Consider adding unit tests for all the flows that perform background refresh, not just client_credentials:

- refresh_token
- obo
- msi